### PR TITLE
Add ability to have skewed shared counter traffic

### PR DESCRIPTION
**Pull request by sadhansood**
This PR allows us to create hot shared object workload if we want to in load gen. Previously, we created a unique shared counter to go with every request but with this we now first create a pool of shared counters which are reused across requests. Also added a tuning parameter `shared_counter_hotness_factor` which let us control how many unique shared counters will be created. A value of 100 i.e. maximum hotness will direct all traffic to just one shared counter whereas default of 0 will have each request use a different counter (current behavior)
Running on my local machine will non zero hotness leads to a bunch of logs like:
```
The store does not contain the parent of msqMEXjB5Lgj7VmE: C948(jGbcLB6p5T8JhcF7TnrxmRK208QODFkgpaElCbTrNhn14H7Fbqd/CzBim6HMctdbE5RgeCpfDi+J+0xCtLil+uPSYBAiIOY9B1Tn4YRt7v05iOreTtN/E4VDfRneGhYY, WO8DtNfwbEkQMEAM, E0): Missing item digest=bNuHJo320/i2hD1RrYU2fTTq5wnB73ByMWq9QB2Ojzo=
Throughput = 115.488884, latency_ms(min/p50/p99/max) = 494/831/1631/1791, num_success = 1050, num_error = 0, no_gas = 0, submitted = 999, in_flight = 85
2022-11-09T23:53:36.527624Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of sPryyxi706aqVwK3: C1016(i9DpjC/kZbDd57csSg7qWFMF3uTECO2jBD3M5BOXU8T7AebTrMO+9D5IbnUI8N7lFhmjEiKFhwUOlFGbx1jkdP212PR/XvnayHxDaJuMkytxACBnshi9TrbZiZP10tqL, mV7uAKc6hlEs7dGc, E0): Missing item digest=9T+a2mWrgBGok65wUmZU+4i06SA+kC/unDlWPkP25sA=
2022-11-09T23:53:36.530187Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of sPryyxi706aqVwK3: C1016(i9DpjC/kZbDd57csSg7qWFMF3uTECO2jBD3M5BOXU8T7AebTrMO+9D5IbnUI8N7lFhmjEiKFhwUOlFGbx1jkdP212PR/XvnayHxDaJuMkytxACBnshi9TrbZiZP10tqL, mV7uAKc6hlEs7dGc, E0): Missing item digest=9T+a2mWrgBGok65wUmZU+4i06SA+kC/unDlWPkP25sA=
2022-11-09T23:53:36.540587Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of sPryyxi706aqVwK3: C1016(i9DpjC/kZbDd57csSg7qWFMF3uTECO2jBD3M5BOXU8T7AebTrMO+9D5IbnUI8N7lFhmjEiKFhwUOlFGbx1jkdP212PR/XvnayHxDaJuMkytxACBnshi9TrbZiZP10tqL, mV7uAKc6hlEs7dGc, E0): Missing item digest=9T+a2mWrgBGok65wUmZU+4i06SA+kC/unDlWPkP25sA=
2022-11-09T23:53:40.486578Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of BodtifHuP/raBxDt: C1048(trcpKg/G/U0ea9SlxtbZOTc7FKdaVfe5s6jn+EWYhzrh0Wwv+zEyLF12WaiCaOlkAqCUH//2qnxMMX+YTaWFqZ7hqL7Rk2jxlO7atvXvTuxprDCAFY7WWR8t64m8GwW/, 8thZmnG4u95Wdiws, E0): Missing item digest=OBdk/VZuT9wpV3BKkjq22ROjfRzxrITU+qyQq2cBPFI=
2022-11-09T23:53:40.488631Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of BodtifHuP/raBxDt: C1048(trcpKg/G/U0ea9SlxtbZOTc7FKdaVfe5s6jn+EWYhzrh0Wwv+zEyLF12WaiCaOlkAqCUH//2qnxMMX+YTaWFqZ7hqL7Rk2jxlO7atvXvTuxprDCAFY7WWR8t64m8GwW/, 8thZmnG4u95Wdiws, E0): Missing item digest=OBdk/VZuT9wpV3BKkjq22ROjfRzxrITU+qyQq2cBPFI=
2022-11-09T23:53:40.490573Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of BodtifHuP/raBxDt: C1048(trcpKg/G/U0ea9SlxtbZOTc7FKdaVfe5s6jn+EWYhzrh0Wwv+zEyLF12WaiCaOlkAqCUH//2qnxMMX+YTaWFqZ7hqL7Rk2jxlO7atvXvTuxprDCAFY7WWR8t64m8GwW/, 8thZmnG4u95Wdiws, E0): Missing item digest=OBdk/VZuT9wpV3BKkjq22ROjfRzxrITU+qyQq2cBPFI=
2022-11-09T23:53:40.507035Z ERROR narwhal_consensus::bullshark: The store does not contain the parent of BodtifHuP/raBxDt: C1048(trcpKg/G/U0ea9SlxtbZOTc7FKdaVfe5s6jn+EWYhzrh0Wwv+zEyLF12WaiCaOlkAqCUH//2qnxMMX+YTaWFqZ7hqL7Rk2jxlO7atvXvTuxprDCAFY7WWR8t64m8GwW/, 8thZmnG4u95Wdiws, E0): Missing item digest=OBdk/VZuT9wpV3BKkjq22ROjfRzxrITU+qyQq2cBPFI=
```

### CODE REVIEW

File: crates/sui-benchmark/src/bin/stress.rs [hunk 1]
```diff
@@ -404,8 +410,11 @@ async fn main() -> Result<()> {
                     stat_collection_interval,
                     shared_counter,
                     transfer_object,
+                    shared_counter_hotness_factor,
                     ..
                 } => {
+                    let shared_counter_ratio = 1.0
+                        - (std::cmp::min(shared_counter_hotness_factor as u32, 100) as f32 / 100.0);
```
Comment (mystenmark) [inline crates/sui-benchmark/src/bin/stress.rs:417]: it probably doesn't matter but you may as well use f64 here.

**Review by mystenmark — APPROVED** [review]
thanks for this! I think this is fine as is but I find the "hotness factor" to be slightly confusing. I take it it is the inverse ratio of total gas objects available for cert formation to the number of unique shared objects? If you can add a comment clarifying that it may be helpful to future readers. Otherwise, LGTM

**Review by mwtian — APPROVED** [review]
Great!

The error log has appeared since it was added. I was trying to find time to debug that.

File: crates/sui-benchmark/src/bin/stress.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-benchmark/src/bin/stress.rs:417]: sure thing, let me add another comment explaining it better

**Pull request merged**